Furball.Vixie/Graphics/Backends/OpenGL20/OpenGL20Backend.cs

```diff
@@ -0,0 +1,141 @@
+            Logger.Log($"Renderer:       {this.gl.GetStringS(StringName.Renderer)}",               LoggerLevelOpenGL20.InstanceInfo);
+
+            this._imgui = new ImGuiController(this.gl, Global.GameInstance.WindowManager.GameWindow, Global.GameInstance._inputContext);
+        }
+        
+        /// <summary>
+        /// Debug Callback
+        /// </summary>
+        private void Callback(GLEnum source, GLEnum type, int id, GLEnum severity, int length, nint message, nint userparam) {
+            string stringMessage = SilkMarshal.PtrToString(message);
+
+            LoggerLevel level = severity switch {
+                GLEnum.DebugSeverityHigh         => LoggerLevelDebugMessageCallback.InstanceHigh,
+                GLEnum.DebugSeverityMedium       => LoggerLevelDebugMessageCallback.InstanceMedium,
+                GLEnum.DebugSeverityLow          => LoggerLevelDebugMessageCallback.InstanceLow,
+                GLEnum.DebugSeverityNotification => LoggerLevelDebugMessageCallback.InstanceNotification,
+                _                                => null
+            };
+
+            Console.WriteLine(stringMessage);
+        }
+        
+        public override void Cleanup() {
+            this.gl.Dispose();
+        }
+
+        public override void HandleWindowSizeChange(int width, int height) {
+            this.gl.Viewport(0, 0, (uint)width, (uint)height);
+            
+            this.gl.MatrixMode(MatrixMode.Projection);
+            this.gl.LoadIdentity();
+            this.gl.Ortho(0, width, height, 0, 0, 1);
+            
+            this.ProjectionMatrix = Matrix4x4.CreateOrthographicOffCenter(0, width, height, 0, 0, 1);
+        }
+        
+        public override void HandleFramebufferResize(int width, int height) {
+            this.gl.Viewport(0, 0, (uint)width, (uint)height);
+            
+            this.gl.MatrixMode(MatrixMode.Projection);
+            this.gl.LoadIdentity();
+            this.gl.Ortho(0, width, height, 0, 0, 1);
+
+            this.ProjectionMatrix = Matrix4x4.CreateOrthographicOffCenter(0, width, height, 0, 0, 1);
+        }
+        
+        public override IQuadRenderer CreateTextureRenderer() => new QuadRendererGL20(this);
+        public override ILineRenderer CreateLineRenderer()    => new LineRendererGL20(this);
+
+        private         int _maxTexUnits = -1;
+        public override int QueryMaxTextureUnits() {
+            if (this._maxTexUnits == -1)
+                this._maxTexUnits = this.gl.GetInteger((GLEnum)GetPName.MaxTextureImageUnits);
+
+            return this._maxTexUnits;
+        }
+        public override void Clear() {
+            this.gl.Clear(ClearBufferMask.ColorBufferBit);
+            this.gl.ClearColor(0.5f, 0, 0, 0);
+        }
+        public override TextureRenderTarget CreateRenderTarget(uint width,     uint height)      => new TextureRenderTargetGL20(this, width, height);
```

Comment:
this autoformatting looks horrific

Reply:
i think it looks fine but sure lemme change it